packages/rln/src/codec.ts

```diff
@@ -0,0 +1,130 @@
+import type {
```

Comment:
nit: let's split this file into two: `Decoder` and `Encoder` -- it's quite long as is